doc/source/workflows/checkpointing.rst

```diff
@@ -0,0 +1,257 @@
+Checkpointing
+=============
+
+Introduction
+------------
+
+Ray Workflows provides strong fault tolerance and exactly-once execution semantics by checkpointing the workflow DAG before execution and the outputs after execution. However, checkpointing is time consuming, especially when you have large inputs and outputs for workflow steps. When exactly-once execution semantics is not required, you can skip some checkpoints to speed up your workflow.
+
+Besides skipping checkpoints, an alternative is to replace workflow steps with Ray remote functions. But there are 3 reasons against it:
+
+1. You cannot compose a workflow DAG by mixing workflow steps and Ray remote functions.
+2. Unlike workflow steps, Ray workflow would not display progress and status of Ray remote functions.
+3. The remote functions won’t be considered as part of the workflow, so they won’t be recovered automatically when they fail. For example, instead of returning a workflow (step) B in workflow step A, you return a ObjectRef of a remote function. With workflow step B, workflow enables recovery from step B; but with the remote function, the workflow can only recover from step A.
```

Comment:
I would skip this, probably too much detail for a docs page.

Reply:
I agree. Should we just delete them for now?

Reply:
Yeah actually I think it would be best to not include docs right now. I think one of us can take a pass on this in a separate PR.